test/extended/apiserver/resiliency.go

```diff
@@ -94,6 +94,10 @@ var _ = ginkgo.Describe("[Conformance][sig-sno][Serial] Cluster", func() {
 		ginkgo.By("with no pods restarts during API disruption")
 		names := GetRestartedPods(c, restartingContainers)
 		gomega.Expect(len(names)).To(gomega.Equal(0), "Some pods in got restarted during kube-apiserver rollout: %s", strings.Join(names, ", "))
+
+		// Workaround for issues identified in https://bugzilla.redhat.com/show_bug.cgi?id=2059581
+		// TODO: Remove this sleep once that bug is resolved
+		time.Sleep(60 * time.Second)
```

Comment:
maybe we can rather wait (for the same amount of time) with some condition to check api availability?

Reply:
another option is to check api availability right after this sleep duration (I prefer the first option, but with this we at least know if the sleep helped or not)

Reply:
>maybe we can rather wait (for the same amount of time) with some condition to check api availability?
This is already what the loop is doing. The issue is that during the period after the test, the API has fluctuating availability. It's available for a few seconds, then it isn't, then it's available again.

This is what the original test loop is already doing... checking the availability. The issue is that the API is reporting ready while it's not actually ready and has fluctuating availability, for about 10 seconds. We want to skip that period. It usually take like 10 seconds, we wait well beyond it, 60 seconds

Reply:
Ack. I didn't understand that it has fluctuating availability in that period of time

Reply:
Is the suggestion from @omertuc satisfactory or do we still need further actions here before we can merge?